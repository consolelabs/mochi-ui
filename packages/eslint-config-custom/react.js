const { resolve } = require('node:path')

const project = resolve(process.cwd(), 'tsconfig.json')

const dfLintConfig = require.resolve('@dwarvesf/react-eslint-config')

module.exports = {
  extends: [dfLintConfig, 'eslint-config-turbo'],
  plugins: ['prettier'],
  parserOptions: {
    project,
  },
  globals: {
    JSX: true,
  },
  ignorePatterns: [
    'node_modules/',
    'dist/',
    '.eslintrc.js',
    '**/*.css',
    '*.snap',
  ],
  settings: {
    'import/resolver': {
      typescript: {
        project,
      },
      node: {
        extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
  rules: {
    'react/function-component-definition': 'off',
    'turbo/no-undeclared-env-vars': 'off',
    'react/jsx-key': 'off',
    'no-dupe-class-members': 'off',
    'prefer-arrow-callback': 'off',
    'import/export': 'off',
    'prettier/prettier': [
      'error',
      {
        bracketSpacing: true,
        jsxBracketSameLine: false,
        jsxSingleQuote: false,
        printWidth: 80,
        proseWrap: 'always',
        semi: false,
        singleQuote: true,
        tabWidth: 2,
        trailingComma: 'all',
        xmlWhitespaceSensitivity: 'ignore',
      },
    ],
  },
  overrides: [
    {
      parser: '@typescript-eslint/parser',
      files: ['**/*.ts', '**/*.tsx'],
      plugins: ['@typescript-eslint'],
      rules: {
        'no-use-before-define': 'off',
        'react/react-in-jsx-scope': 'off',
        'react/jsx-sort-props': 'off',
        'no-void': 'off',
        'no-unused-vars': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            argsIgnorePattern: '^_',
            varsIgnorePattern: '^_',
            caughtErrorsIgnorePattern: '^_',
          },
        ],
        '@typescript-eslint/no-use-before-define': [
          'error',
          { functions: false, classes: false },
        ],
      },
    },
  ],
}
