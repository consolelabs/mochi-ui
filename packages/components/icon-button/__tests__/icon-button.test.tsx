import { render, fireEvent } from '@testing-library/react'
import IconButton from '../src/icon-button'

describe('IconButton', () => {
  it('renders correctly with children', () => {
    const { container } = render(<IconButton>Click me</IconButton>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with variant', () => {
    const { container } = render(
      <IconButton variant="outline">Click me</IconButton>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with color', () => {
    const { container } = render(
      <IconButton color="primary">Click me</IconButton>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with size', () => {
    const { container } = render(<IconButton size="lg">Click me</IconButton>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly when disabled', () => {
    const { container } = render(<IconButton disabled>Click me</IconButton>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('calls onClick when clicked', () => {
    const handleClick = jest.fn()
    const { getByRole } = render(
      <IconButton onClick={handleClick}>Click me</IconButton>,
    )
    fireEvent.click(getByRole('button'))
    expect(handleClick).toHaveBeenCalled()
  })

  it('renders correctly with custom class name', () => {
    const { container } = render(
      <IconButton className="custom-class">Click me</IconButton>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with additional props', () => {
    const { container } = render(
      <IconButton data-testid="my-button">Click me</IconButton>,
    )
    expect(container.firstChild).toHaveAttribute('data-testid', 'my-button')
  })
})
