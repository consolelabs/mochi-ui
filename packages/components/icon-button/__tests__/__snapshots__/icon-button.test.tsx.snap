// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly when disabled 1`] = `
<button
  class="flex items-center gap-x-2 rounded-full font-semibold h-fit shadow-button border text-sm cursor-not-allowed p-2.5 bg-primary-700 text-white border-transparent hover:bg-primary-800 focus:shadow-button-focused-primary disabled:bg-primary-400"
  disabled=""
  type="button"
>
  Click me
</button>
`;

exports[`IconButton renders correctly with children 1`] = `
<button
  class="flex items-center gap-x-2 rounded-full font-semibold h-fit shadow-button border text-sm p-2.5 bg-primary-700 text-white border-transparent hover:bg-primary-800 focus:shadow-button-focused-primary disabled:bg-primary-400"
  type="button"
>
  Click me
</button>
`;

exports[`IconButton renders correctly with color 1`] = `
<button
  class="flex items-center gap-x-2 rounded-full font-semibold h-fit shadow-button border text-sm p-2.5 bg-primary-700 text-white border-transparent hover:bg-primary-800 focus:shadow-button-focused-primary disabled:bg-primary-400"
  type="button"
>
  Click me
</button>
`;

exports[`IconButton renders correctly with custom class name 1`] = `
<button
  class="flex items-center gap-x-2 rounded-full font-semibold h-fit shadow-button border text-sm p-2.5 bg-primary-700 text-white border-transparent hover:bg-primary-800 focus:shadow-button-focused-primary disabled:bg-primary-400 custom-class"
  type="button"
>
  Click me
</button>
`;

exports[`IconButton renders correctly with size 1`] = `
<button
  class="flex items-center gap-x-2 rounded-full font-semibold h-fit shadow-button border text-base p-3 bg-primary-700 text-white border-transparent hover:bg-primary-800 focus:shadow-button-focused-primary disabled:bg-primary-400"
  type="button"
>
  Click me
</button>
`;

exports[`IconButton renders correctly with variant 1`] = `
<button
  class="flex items-center gap-x-2 rounded-full font-semibold h-fit shadow-button border text-sm p-2.5 bg-white text-primary-700 border-primary-300 hover:bg-primary-100 focus:shadow-button-focused-primary disabled:text-primary-400 disabled:bg-white"
  type="button"
>
  Click me
</button>
`;
